tests/resources/ci/scripts/setup.sh

```diff
@@ -91,28 +146,48 @@ installGradle() {
     local gradleHome="${SOFTWARE_INSTALL_DIR}/gradle-${GRADLE_VERSION}"
 	local url="https://services.gradle.org/distributions/gradle-${GRADLE_VERSION}-bin.zip"
 
-	# Download, validate, and expand the Maven archive.
-	curl -fsSL -o /tmp/liberty-dev-tool-gradle.zip "$url"
-	echo "${GRADLE_ARCHIVE_SHA256}  /tmp/liberty-dev-tool-gradle.zip" | sha256sum -c - 
+    # Download the Gradle archive.
+    curl -fsSL -o /tmp/liberty-dev-tool-gradle.zip "$url"
+      
+    # Check the downloaded archive's SHA against the expected value.
+	if [[ $OS == "Linux" ]]; then
+        echo "${GRADLE_ARCHIVE_SHA256}  /tmp/liberty-dev-tool-gradle.zip" | sha256sum -c -
+    elif [[ $OS == "Darwin" ]]; then
+        echo "${GRADLE_ARCHIVE_SHA256}  /tmp/liberty-dev-tool-gradle.zip" | shasum -a 256 -c -
+    else
+        local shaAll=$(certutil -hashfile /tmp/liberty-dev-tool-gradle.zip SHA256)
+        local downloadedZipSha=$(echo $(echo $shaAll | tr '\r' ' ') | cut -d " " -f 5)
+        if [ "$GRADLE_ARCHIVE_SHA256" != "$downloadedZipSha" ]; then
+            echo "ERROR: expected SHA: $GRADLE_ARCHIVE_SHA256 is not equal to downloaded file calculated SHA of: $downloadedZipSha"
+            exit -1
+        fi
+    fi
+
+    # Expand the archive.
     unzip -d "$SOFTWARE_INSTALL_DIR" /tmp/liberty-dev-tool-gradle.zip
-
+    
     # Set the GRADLE_HOME environment variable and make it available to other steps within the executing job.
     echo "GRADLE_HOME=${gradleHome}" >> $GITHUB_ENV
 
     # Prepend the Gradle installation's bin dir location to PATH and make it available to other steps within the executing job.
     echo "${gradleHome}/bin" >> $GITHUB_PATH
 }
 
+# installXDisplaySoftwareOnLinux Installs a X display, a windows manager, and other pre-req software.
+installXDisplaySoftwareOnLinux() {
+    sudo apt-get install dbus-x11 xvfb metacity at-spi2-core 
+} 
+
 # installDocker installs docker.
-installDocker() {
+installDockerOnLinux() {
```

Comment:
Just curious:  what do we install docker for ?

Reply:
As i mentioned before, there is at least one more function test that needs to be written. That is the test that starts the app in devmode on a docker container. That is just adding the needed infrastructure ahead of the test.

Reply:
Ah..got it..thx